tests/test_analysis.py

```diff
@@ -420,50 +420,68 @@ def test_mix_scenarios():
     }
 
 
-def test_scenario_switcher(seed):
+def test_improved_healthsystem_and_care_seeking_scenario_switcher(seed):
```

Comment:
This is a great test but I feel like its not fully testing our expectation from the scenario switcher. If I have understood PR https://github.com/UCL/TLOmodel/pull/1326 quite well, the intention of this module/class is to cause changes(updating parameters) during simulation(yearly?) and not only at the start or end of the simulation. if thats the case then I could love it the more if this test could demonstrate that by  creating a dummy module with an event that runs `check_parameters` yearly for some few years and `phase_of_simulation` argument not manually set rather taken from scenario switcher. Don't know if this makes sense?

Reply:
I see what you mean and think you make a good point. Let me see if I can make a better test.

Reply:
Next commit I think does this! :-)